daemon/src/request/states.rs

```diff
@@ -102,13 +105,24 @@ impl Status {
     /// progress. And finally, the last case is that both `Status`es agree
     /// the `Status` is `Available`.
     #[must_use]
-    pub const fn join(self, other: Self) -> Self {
-        match (self, other) {
-            (Self::Failed, _) | (_, Self::Failed) => Self::Failed,
+    pub fn join(&self, other: &Self) -> Self {
+        match (self, &other) {
+            (Self::Failed { reason: reason1 }, Self::Failed { reason: reason2 }) => Self::Failed {
+                reason: format!("{} and {}", reason1, reason2),
```

Comment:
Ah, but it might be more complicated here. Can we chain reasons?

Reply:
I'm not sure I know what you mean? Something like elaborate the type so that it can have multiple reason fields rather than concatenating strings?

Reply:
Sorry, this was in relation to this comment https://github.com/radicle-dev/radicle-link/pull/651#discussion_r626638113. So this can be resolved :)